implementation/orchestration-to-choreography/README.md

```diff
@@ -0,0 +1,90 @@
+
+The code will deploy 2 CloudFormation stacks:
+- OrchestrationStack: Here is the State Machine that invokes a number of Lambda functions, coordinates the communication, and completes the workflow.
+- ChoreographyStack: Here are a number of tasks (in the form of Lambda function), which operate independently without coupling with each other. They use Publisher/Subscriber pattern and message to queue to work together asynchronously and ultimately achieve the same result as an explicit workflow.
+
+---
+## Deploy the infrastructure
+
+
+To build this app, navigate to `implementation/orchestration-to-choreography` folder. Then run the following:
+
+```bash
+npm install -g aws-cdk
+npm install
+```
+
+This will install the necessary CDK, dependencies, build your TypeScript files and CloudFormation template.
+
+Next, deploy the 2 Stacks to your AWS Account (`OrchestrationStack` and `ChoreographyStack`):
+``` 
+cdk deploy --all
+```
+
+
+## Testing it out
+
+1. First, in the orchestration scenario, the coordinator needs to know all the involved parties (e.g. the list of banks). Let's create a record in DynamoDB to provide the bank list. You can do this in AWS Console or CLI. The CLI command is shown as below:
+ ```
+aws dynamodb put-item --table-name LoanBrokerBanks --item '{"Type": {"S": "Home"}, "BankAddress": {"L": [{"S": "BankRecipientPawnshop"}, {"S": "BankRecipientUniversal"}, {"S": "BankRecipientPremium"}]}}'
+```
+
+2. Next, we can perform the Step Functions State Machine:
+    - Go to the AWS Console and then to Step Functions
+    - Click on the LoanBroker State Machine
+    - Click the Start Execution button and enter this input: `{"SSN": "123-45-6789", "Amount": 400000, "Term": 30 }`
+    - After a few seconds, all the states should turn green.
+    - Click on the Map state, and choose "View Map state overview", in the "Output" tab, you should be able to see the bank offers.
+ ```
+{
+  "SSN": "123-45-6789",
+  "Amount": 400000,
+  "Term": 30,
+  "Credit": {
+    "Score": 511,
+    "History": 25
+  },
+  "Banks": {
+    "BankAddress": [
+      "BankRecipientPawnshop",
+      "BankRecipientUniversal",
+      "BankRecipientPremium"
+    ]
+  },
+  "Quotes": [
+    {
+      "Quote": {
+        "rate": 8.180046213479706,
+        "bankId": "PawnShop"
+      }
+    },
+    ...
+  ]
+}
+ ```
+
+3. Now, let's test the choreagraphy scenario and validate it can achieve the same functionality. We issue a message to SNS topic to trigger the distributed processing. At the end, the aggregated quote result is written into DynamoDB by an aggregation Lambda function.
+ ```
+aws sns publish --topic-arn arn:aws:sns:<region>:<AccountId>:MortgageQuoteRequest \
+  --message '{ "SSN": "123-45-6666", "Amount": 500000, "Term": 30, "Credit": { "Score": 803, "History": 22 } }'
```

Comment:
We want to match the functionality in the 'before' and 'after'.  
Looks like in the 'after' implementation you are skipping the step "Get Credit Score" and directly passing score in the input to SNS.

Reply:
Added "Get Credit Score" and a state machine to issue message to SNS topic.